src/App.tsx

```diff
@@ -31,65 +31,115 @@ export function getReorderedGoods(
   goods: string[],
   { sortType, isReversed }: ReorderOptions,
 ) {
-  // To avoid the original array mutation
-  const visibleGoods = [...goods];
+  let visibleGoods = [...goods];
+
+  if (sortType === SortType.ALPHABET) {
+    visibleGoods = visibleGoods.sort((a, b) => {
+      return a.localeCompare(b);
+    });
+  }
+
+  if (sortType === SortType.LENGTH) {
+    visibleGoods = visibleGoods.sort((a, b) => {
+      return a.length - b.length;
+    });
+  }
+
+  if (isReversed) {
+    visibleGoods = visibleGoods.reverse();
```

Comment:
`reverse` mutates the array, pick your fighter ;)

```suggestion
    visibleGoods.reverse();
    
    // OR
    
    visibleGoods = [...visibleGoods].reverse()
```